src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -315,6 +345,161 @@ private void ResolveDotNetDesktopAsymmetricAlgorithm(AsymmetricSecurityKey key,
         }
 #endif
 
+        private RSAParameters CreateRsaParametersFromJsonWebKey(JsonWebKey webKey, bool willCreateSignatures)
+        {
+            RSAParameters parameters;
+            if (willCreateSignatures)
+            {
+                parameters = new RSAParameters()
+                {
+                    D = Base64UrlEncoder.DecodeBytes(webKey.D),
```

Comment:
Is `Base64UrlEncoder.DecodeBytes` guarded against null values?
A malformed JWKS response might miss one of these precious parameters and we should have a nice exception path for that.